matrix_sdk_crypto/src/olm.rs

```diff
@@ -94,7 +94,7 @@ impl Account {
 
         Account {
             user_id: Arc::new(user_id.to_owned()),
-            device_id: Arc::new(device_id.to_owned()),
+            device_id: Arc::new(device_id.to_owned().into_boxed_str()),
```

Comment:
```suggestion
            device_id: Arc::new(device_id.into()),
```

[should work](https://doc.rust-lang.org/std/boxed/struct.Box.html#impl-From%3C%26%27_%20str%3E)

Reply:
Same for all other `.to_owned().into_boxed_str()` / `device_id_ref.to_string().into_boxed_str()`.